java-checks/src/main/java/org/sonar/java/checks/FieldsShouldNotBeInitializedToDefaultValuesCheck.java

```diff
@@ -0,0 +1,90 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.List;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.semantic.Type.Primitives;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.tree.VariableTree;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+@Rule(
+  key = "S3052",
+  name = "Fields should not be initialized to default values",
+  tags = {"convention"},
+  priority = Priority.MINOR)
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
+@SqaleConstantRemediation("2min")
+public class FieldsShouldNotBeInitializedToDefaultValuesCheck extends SubscriptionBaseVisitor {
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.CLASS, Tree.Kind.ENUM);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    ClassTree classTree = (ClassTree) tree;
+    for (Tree member : classTree.members()) {
+      if (member.is(Tree.Kind.VARIABLE)) {
+        checkField((VariableTree) member);
+      }
+    }
+  }
+
+  public void checkField(VariableTree fieldTree) {
+    // No initialization or not literal => go out
+    if (!(fieldTree.initializer() instanceof LiteralTree)) {
+      return;
+    }
+
+    boolean explicitInit = false;
+    LiteralTree initTree = (LiteralTree) fieldTree.initializer();
+    try {
+      if (!initTree.symbolType().isPrimitive() && "null".equals(initTree.value())) {
+        explicitInit = true;
+      } else if (initTree.symbolType().isPrimitive() && initTree.symbolType().isNumerical()
+        && NumberFormat.getInstance().parse(initTree.value()).shortValue() == 0) {
+        explicitInit = true;
+      } else if (initTree.symbolType().isPrimitive(Primitives.BOOLEAN) && !Boolean.parseBoolean(initTree.value())) {
+        explicitInit = true;
+      }
+    } catch (ParseException e) {
```

Comment:
I am not a found of surrounding the whole else if with a try catch for only some cases of this else if.

Moreover, I tend to think that, as this else if is only setting a variable to true, the conditions should be extracted in well labeled methods returning a boolean to express behaviour properly and thus reducing the main code to only one if with a or condition based on those extracted methods.